modules/cudev/include/opencv2/cudev/grid/detail/reduce_to_column.hpp

```diff
@@ -101,7 +101,7 @@ namespace grid_reduce_to_vec_detail
 
         __shared__ work_elem_type smem[cn][BLOCK_SIZE];
 
-        const int y = blockIdx.x;
+        const int y = blockIdx.x*cols;
```

Comment:
In generic case `dst` matrix will not be contiguous, so this index is not valid.

Original solution (`dst.create(1, rows);`) was used to create contiguous matrix, so that it can be accessed by 1D index.